livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -407,6 +408,13 @@ private String callRestAPI(String targetURL, String method, String jsonData)
       response = new ResponseEntity(e.getResponseBodyAsString(), e.getStatusCode());
       LOGGER.error(String.format("Error with %s StatusCode: %s",
           response.getStatusCode().value(), e.getResponseBodyAsString()));
+    } catch (RestClientException e) {
+      // Exception happens when kerberos is enabled.
+      if (e.getMessage().contains("404 Not Found")) {
+        throw new SessionNotFoundException("Session not found when kerberos is enabled ");
```

Comment:
right, it has the stack but not the exception type.
it just seems somewhat generic - I assume RestClientException could be possible in other error cases too? Is there an inner exception that we could use to scope this down?